Google.PowerShell/Compute/GceHealthCheckCmdlets.cs

```diff
@@ -0,0 +1,545 @@
+        /// <para type="description">
+        /// If set, will remove a health check that uses http.
+        /// </para>
+        /// </summary>
+        [Parameter(ParameterSetName = ParameterSetNames.ByNameHttp, Mandatory = true)]
+        [Parameter(ParameterSetName = ParameterSetNames.ByHttpObject)]
+        public SwitchParameter Http { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// If set, will remove a health check that uses https.
+        /// </para>
+        /// </summary>
+        [Parameter(ParameterSetName = ParameterSetNames.ByNameHttps, Mandatory = true)]
+        [Parameter(ParameterSetName = ParameterSetNames.ByHttpsObject)]
+        public SwitchParameter Https { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// An object defining the http health check to remove.
+        /// </para>
+        /// </summary>
+        [Parameter(ParameterSetName = ParameterSetNames.ByHttpObject, Mandatory = true,
+            Position = 0, ValueFromPipeline = true)]
+        public HttpHealthCheck HttpObject { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// An object defining the https health check to remove.
+        /// </para>
+        /// </summary>
+        [Parameter(ParameterSetName = ParameterSetNames.ByHttpsObject, Mandatory = true,
+            Position = 0, ValueFromPipeline = true)]
+        public HttpsHealthCheck HttpsObject { get; set; }
+
+        protected override void ProcessRecord()
+        {
+            switch (ParameterSetName)
+            {
+                case ParameterSetNames.ByNameHttp:
+                    RemoveHttpHealthCheck(Project, Name);
+                    break;
+                case ParameterSetNames.ByNameHttps:
+                    RemoveHttpsHealthCheck(Project, Name);
+                    break;
+                case ParameterSetNames.ByHttpObject:
+                    RemoveHttpHealthCheck(GetProjectNameFromUri(HttpObject.SelfLink), HttpObject.Name);
+                    break;
+                case ParameterSetNames.ByHttpsObject:
+                    RemoveHttpsHealthCheck(GetProjectNameFromUri(HttpsObject.SelfLink), HttpsObject.Name);
+                    break;
+                default:
+                    throw UnknownParameterSetException;
+            }
+        }
+
+        private void RemoveHttpsHealthCheck(string project, string name)
+        {
+            if (ShouldProcess($"{project}/{name}", "Remove Https Health Check"))
+            {
+                HttpsHealthChecksResource.DeleteRequest request = Service.HttpsHealthChecks.Delete(project, name);
+                Operation operation = request.Execute();
+                AddGlobalOperation(project, operation);
+            }
+        }
+
+        private void RemoveHttpHealthCheck(string project, string name)
+        {
+            if (ShouldProcess($"{project}/{name}", "Remove Http Health Check"))
```

Comment:
When I saw the integration tests, I saw we didn't need to pass `-Confirm` or `-Force`. I assume this is because Pester auto-confirms prompts when we call `ShouldProcess`?